### PR TITLE
Make sure that machete file for worktrees is sought for in the top-level git dir

### LINKED ISSUE #360: Support for worktrees should NOT be identical to the support for submodules
As noted by @kgadek:

```
❯ git worktree add ../foobar some-branch
Preparing worktree (checking out 'some-branch')
HEAD is now at 93fa883a .....

❯ cd ../foobar

❯ git machete status --list-commits-with-hashes

No branches listed in /Users/kgadek/src/some-project/.git/worktrees/foobar/machete; use git machete discover or git machete edit, or edit /Users/kgadek/src/some-project/.git/worktrees/foobar/machete manually.

❯ file .git
.git: ASCII text

❯ cat .git
gitdir: /Users/kgadek/src/some-project/.git/worktrees/foobar
```

This is exactly what happens for submodules (`machete` file is sought for in per-submodule `.git/` folder, and NOT in the top-project `.git/` folder)... but given the different nature of worktrees, we should rather look for `machete` file in top-project `.git/` folder (so `/Users/kgadek/src/some-project/.git/machete` and not `/Users/kgadek/src/some-project/.git/worktrees/foobar/machete`).

TBD how to elegantly distinguish between the submodule and worktree case (probably just whether the git path ends with `/.git/worktrees/<folder>`?)

Also, important to make sure that git compatibility is preserved (shouldn't be very difficult tho).

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: docs/source/cli_help/file.rst [hunk 1]
```diff
@@ -10,3 +10,11 @@ file
 
 Outputs the absolute path of the machete definition file. Currently fixed to ``<git-directory>/machete``.
 Note: this won't always be just ``<repo-root>/.git/machete`` since e.g. submodules and worktrees have their git directories in different location.
+
+Outputs the absolute path of the machete definition file.
```
Comment (rifel123) [inline docs/source/cli_help/file.rst]: `Outputs the absolute path of the machete definition file.` doubled? Line 11 and 12 should be removed like in `docs.py`

**Review by rifel123 — APPROVED** [review]

File: docs/source/cli_help/file.rst (hunk 1 above)
Comment (PawelLipski) [reply docs/source/cli_help/file.rst]: Whoops good spot

File: docs/source/cli_help/file.rst [hunk 2]
```diff
@@ -8,5 +8,10 @@ file
...
     git machete file
 
-Outputs the absolute path of the machete definition file. Currently fixed to ``<git-directory>/machete``.
-Note: this won't always be just ``<repo-root>/.git/machete`` since e.g. submodules and worktrees have their git directories in different location.
+Outputs the absolute path of machete definition file.
+The file is always called ``machete`` and is located in the git directory of the project.
+
+Three cases are possible:
+    * if ``git machete`` is executed from a regular working directory (not a worktree or submodule), this simply resolves to `machete` in .git folder,
+    * if `git machete` is executed from a **worktree**, this resolves to `machete` in the .git folder of the **top-level project** (not the worktree's .git folder!),
```
Comment (PawelLipski) [inline docs/source/cli_help/file.rst]: Whoops missing double backticks...

File: docs/source/cli_help/discover.rst [hunk 3]
```diff
@@ -9,7 +9,7 @@ discover
     git machete discover [-C|--checked-out-since=<date>] [-l|--list-commits] [-r|--roots=<branch1>,<branch2>,...] [-y|--yes]
 
 Discovers and displays tree of branch dependencies using a heuristic based on reflogs and asks whether to overwrite the existing definition :ref:`file` with the new discovered tree.
-If confirmed with a ``y[es]`` or ``e[dit]`` reply, backs up the current definition file (if it exists) as ``$GIT_DIR/machete~`` and saves the new tree under the usual ``$GIT_DIR/machete`` path.
+If confirmed with a ``y[es]`` or ``e[dit]`` reply, backs up the current definition file (if it exists) as ``machete~`` and saves the new tree in the usual ``machete`` file.
```
Comment (kgadek) [inline docs/source/cli_help/discover.rst]: TBH not sure if this is more clear than the original

File: docs/source/cli_help/file.rst [hunk 4]
```diff
@@ -8,5 +8,10 @@ file
 
     git machete file
 
-Outputs the absolute path of the machete definition file. Currently fixed to ``<git-directory>/machete``.
-Note: this won't always be just ``<repo-root>/.git/machete`` since e.g. submodules and worktrees have their git directories in different location.
+Outputs the absolute path of machete definition file.
+The file is always called ``machete`` and is located in the git directory of the project.
+
+Three cases are possible:
+    * if ``git machete`` is executed from a regular working directory (not a worktree or submodule), this simply resolves to ``machete`` in .git folder,
```
Comment (kgadek) [inline docs/source/cli_help/file.rst]:
```suggestion
    * if ``git machete`` is executed from a regular working directory (not a worktree or submodule), this simply resolves to ``.git/machete``
```

File: docs/source/cli_help/file.rst [hunk 5]
```diff
@@ -8,5 +8,10 @@ file
...
+    * if ``git machete`` is executed from a **worktree**, this resolves to ``machete`` in the .git folder of the **top-level project** (not the worktree's .git folder!),
```
Comment (kgadek) [inline docs/source/cli_help/file.rst]: Technically speaking, in the worktree the `.git` is a file. Moral equivalent of a symlink

File: git_machete/docs.py [hunk 6]
```diff
@@ -208,8 +208,13 @@
...
 
-        Outputs the absolute path of the machete definition file. Currently fixed to `<git-directory>/machete`.
-        Note: this won't always be just `<repo-root>/.git/machete` since e.g. submodules and worktrees have their git directories in different location.
+        Outputs the absolute path of machete definition file.
+        The file is always called `machete` and is located in the git directory of the project.
+
+        Three cases are possible:
+        * if `git machete` is executed from a regular working directory (not a worktree or submodule), this simply resolves to `machete` in .git folder,
+        * if `git machete` is executed from a <b>worktree</b>, this resolves to `machete` in the .git folder of the <b>top-level project</b> (not the worktree's .git folder!),
+        * if `git machete` is executed from a <b>submodule</b>, this resolves to `machete` in the .git folder of the <b>submodule</b> itself (not the top-level project's .git folder!).
```
Comment (kgadek) [inline git_machete/docs.py]: Same comments as for `docs/source/cli_help/file.rst`

**Review by kgadek — APPROVED** [review]
Some 🚲🛖 over documentation, but overall lgtm

File: docs/source/cli_help/discover.rst (hunk 3 above)
Comment (PawelLipski) [reply docs/source/cli_help/discover.rst]: Okay, lemme restore the reference to `$GIT_DIR` (even though it's a mental shortcut)

File: docs/source/cli_help/file.rst (hunk 5 above)
Comment (PawelLipski) [reply docs/source/cli_help/file.rst]: Hmmm I meant top-level's `.git/worktrees/<something>/`! lemme clarify a bit...

**Pull request merged**